smt/surrogate_models/krg_based.py

```diff
@@ -1082,59 +1094,69 @@ def grad_minus_reduced_likelihood_function(log10t):
                     )
 
                     for i in range(len(self.noise0)):
+                        v = i + len(self.theta0)
+                        v2 = deepcopy(v)
```

Comment:
Sorry, I still do not get that one. `deepcopy` should not be needed because in Python interpreter:
```python
>>> a = 1
>>> b = a
>>> a = 2
>>> b
1
>>> a
2
```
Moreover `v` is just used in that deepcopy?!

Reply:
To be honest, I do not know why I have such a problem, what you say seems coherent. 
I will see if I could find another way to make it work.

Reply:
The problem dissapeared. This issue is then closed.